src/Microsoft.DotNet.Interactive.Jupyter.Tests/MagicCommandTests.lsmagic.cs

```diff
@@ -115,5 +118,23 @@ public async Task lsmagic_does_not_list_hidden_commands()
                   .Should()
                   .NotContain("#!hidden");
         }
+
+        [Fact]
+        public async Task r_executes_even_with_comment()
```

Comment:
Rather than this more integrated test, it would make sense to test `DirectiveParser` directly.

Reply:
This makes sense, thanks.

Reply:
You can grab the directive's `Command` directly and parse whatever you like:

![image](https://user-images.githubusercontent.com/547415/116921176-8110a680-ac08-11eb-80c6-217ead76b60e.png)

Reply:
Got it. I currently extract it via reflection, I'll switch to this way then.

Reply:
Although it would require having a kernel, which we wanted to avoid for those tests

Reply:
If there's a refactoring that makes it easier to test this in isolation, that would be useful. Otherwise, I'd prefer to work through the existing public API rather than use reflection, `InternalsVisibleTo`, etc. even if it pulls in a `Kernel`.